Helper/CriteriaHelper.php

```diff
@@ -21,6 +21,46 @@
 class CriteriaHelper
 {
     /**
+     * Generate criterion list to be used to list article
+     *
+     * @param \eZ\Publish\API\Repository\Values\Content\Location $location Location of the folder
+     * @param string[] $excludeContentTypeIdentifiers Array of excluded contentType identifiers
+     * @param string[] $languages Array of languages
+     *
+     * @return \eZ\Publish\API\Repository\Values\Content\Query\Criterion
+     */
+    public function generateListFolderCriterion( Location $location, array $excludeContentTypeIdentifiers = array(), array $languages = array() )
+    {
+        $criteria = array();
+        $criteria[] = new Criterion\Visibility( Criterion\Visibility::VISIBLE );
+        $criteria[] = new Criterion\Subtree( $location->pathString );
```

Comment:
Using `Subtree` instead of `ParentLocationId` will fetch from the whole subtree, I gather this is not intended?